Assignment4/Assignment4/src/Course.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Assignment4
+{
+    public class Course : Event
+    {
+        private string _Subject;
+        //private int _NumStudents;
+
+        private int _NumTeachers = 1;
+
+        private int _RoomNum;
+
+        private static int _CourseTotal;
+
+        public string Subject
+        {
+            get
+            {
+                return _Subject;
+            }
+            set
+            {
+                if(value.Length > 0)
+                {
+                    _Subject = value;
+                }
+                else if(value.Length == 0)
+                {
+                    throw new NotSupportedException("Subject cannot be empty.");           
```

Comment:
I don't think NotSupportiveException is a good choice here. One of the ArgumentExceptions might be more descriptive, because this class _does_ support assigning into the subject
```suggestion
                    throw new ArgumentException("Subject cannot be empty.");           
```